src/mesh/checkpoint_io.C

```diff
@@ -148,11 +150,32 @@ void CheckpointIO::write (const std::string & name)
   // do a gather_to_zero() and support that case too.
   _parallel = _parallel || !mesh.is_serial();
 
+  std::string dir_name;
+  std::string header_file_name = name;
```

Comment:
This `header_file_name` initialization seems unnecessary/should just be moved down a few lines where you are setting it to something else.